catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/controller/BrickController.java

```diff
@@ -0,0 +1,450 @@
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.ui.recyclerview.controller;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import com.google.common.collect.Iterators;
+import com.google.common.collect.PeekingIterator;
+
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.bricks.Brick;
+import org.catrobat.catroid.content.bricks.BrickBaseType;
+import org.catrobat.catroid.content.bricks.ControlStructureBrick;
+import org.catrobat.catroid.content.bricks.IfElseLogicBeginBrick;
+import org.catrobat.catroid.content.bricks.IfLogicBeginBrick;
+import org.catrobat.catroid.content.bricks.IfLogicElseBrick;
+import org.catrobat.catroid.content.bricks.IfLogicEndBrick;
+import org.catrobat.catroid.content.bricks.IfThenLogicBeginBrick;
+import org.catrobat.catroid.content.bricks.IfThenLogicEndBrick;
+import org.catrobat.catroid.content.bricks.LoopBeginBrick;
+import org.catrobat.catroid.content.bricks.LoopEndBrick;
+import org.catrobat.catroid.content.bricks.ScriptBrick;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+
+public final class BrickController {
+
+	private static final String TAG = BrickController.class.getSimpleName();
+
+	@Nullable
+	public Script getScriptThatContainsBrick(Brick brick, @NonNull Iterable<Script> scripts) {
+		for (Script script : scripts) {
+			if (script.containsBrick(brick) || script.getScriptBrick().equals(brick)) {
+				return script;
+			}
+		}
+		return null;
+	}
+
+	public List<Brick> getBricksInControlStructure(ControlStructureBrick controlStructureBrick, @NonNull List<Brick> brickList) {
+		if (!brickList.contains(controlStructureBrick)) {
+			throw new IllegalArgumentException("Brick: " + controlStructureBrick.getClass() + " is not in list.");
+		}
+
+		List<Brick> bricksInControlStructure = new ArrayList<>();
```

Comment:
yes, sure. however, I decided not to use indices (which I would need for `.subList(int .., int ..)` in an attempt to make it safer if there is anything wrong with the control structure. with the iterators (i.e. by not using indices) this won't throw an `ArrayIndexOutOfBoundsException`  instead it will just give you all brick in the list starting from the first `ControlStructureBrick`. I did that because I think in this case the exception does not provide any benefits -> there is no way to fix problems here anyway and besides I think in the case of the Brick that are complicated enough as they are a more tolerant approach is preferable. 

If you or others want to persuade me to use indices I'm happy to hear your arguments :wink: then I would also change  the approach in the `BrickController`.